src/commands/githubActions/endToEndTest/testCheckers.ts

```diff
@@ -116,10 +127,52 @@ async function ensureNoErrorLogs(
 }
 
 /**
- * Test that the validators are attesting after a peri
+ * Test that the validators are attesting after doppelganger protection + 18 minutes max.
  */
-async function attestanceProof(network: Network): Promise<void> {
-  if (network === "mainnet") return;
-  // TODO
-  return;
-}
+export async function attestanceProof(network: Network): Promise<void> {
+  // Wait for doppleganger protection to end, set to 15 min
+  console.log(chalk.grey(`  - Waiting for doppleganger protection to end`));
+  await new Promise(resolve => setTimeout(resolve, 15 * 60 * 1000));
+
+  let response = undefined; // Define response and give it an initial value of undefined
+  let endpoint = "";
+
+  if (network === "mainnet") {
+    // TODO: Add mainnet endpoint
+    // endpoint = `https://beaconcha.in/api/v1/validator/${process.env.VALIDATOR_INDEX}`;
+  } else if (network === "prater") {
+    endpoint = `https://prater.beaconcha.in/api/v1/validator/${process.env.VALIDATOR_INDEX}`;
+  }
+
+  let i = 1;
+  console.log(chalk.grey(`  - Checking if validator is active`));
+
+  while (i <= 9) {
```

Comment:
I would use a for block, which keeps us safe from human errors in the code and having infinite loops